From: Simon McVittie <smcv@debian.org>
Date: Mon, 6 Apr 2020 13:33:09 +0100
Subject: wayland: Guard against nonsense sizes being configured

With the fixes from !1638, it shouldn't be possible for this to happen
any more. However, non-positive sizes make no sense regardless, so if
this does somehow happen, let's make sure *something* reasonable happens.

The practical result of this assertion being hit is that we emit a
critical warning and then behave the same as if !1634 had been merged,
which is known to solve the issue for the submitter.

Signed-off-by: Simon McVittie <smcv@debian.org>
Applied-upstream: 3.24.18
---
 gdk/wayland/gdkwindow-wayland.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gdk/wayland/gdkwindow-wayland.c b/gdk/wayland/gdkwindow-wayland.c
index 6c55544..8502473 100644
--- a/gdk/wayland/gdkwindow-wayland.c
+++ b/gdk/wayland/gdkwindow-wayland.c
@@ -1076,6 +1076,9 @@ gdk_wayland_window_configure (GdkWindow *window,
   GdkDisplay *display;
   GdkEvent *event;
 
+  g_return_if_fail (width > 0);
+  g_return_if_fail (height > 0);
+
   event = gdk_event_new (GDK_CONFIGURE);
   event->configure.window = g_object_ref (window);
   event->configure.send_event = FALSE;
